docs/shared/recipes/nx-release/publish-in-ci-cd.md

```diff
@@ -0,0 +1,207 @@
+
+## Automatically Skip Publishing Locally
+
+When running `nx release`, after the version updates and changelog generation, you will be prompted with the following question:
+
+```{% command="nx release" %}
+...
+? Do you want to publish these versions? (y/N) ›
+```
+
+To move publishing into an automated pipeline, you will want to skip publishing when running `nx release` locally. To do this automatically, use the `--skip-publish` flag:
+
+```{% command="nx release --skip-publish" %}
+...
+
+Skipped publishing packages.
+```
+
+## Use the Publish Subcommand
+
+Nx Release provides a publishing subcommand that performs just the publishing step. Use this in your CI/CD pipeline to publish the packages.
+
+```{% command="nx release publish" %}
+ >  NX   Running target nx-release-publish for 3 projects:
+
+    - pkg-1
+    - pkg-2
+    - pkg-3
+
+ —————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
+
+> nx run pkg-1:nx-release-publish
+
+
+📦  @myorg/pkg-1@0.0.2
+=== Tarball Contents ===
+
+233B README.md
+277B package.json
+53B  src/index.ts
+61B  src/lib/pkg-1.ts
+=== Tarball Details ===
+name:          @myorg/pkg-1
+version:       0.0.2
+filename:      testorg-pkg-1-0.0.2.tgz
+package size:  531 B
+unpacked size: 624 B
+shasum:        {shasum}
+integrity:     {integrity}
+total files:   12
+
+Published to https://registry.npmjs.org with tag "latest"
+
+> nx run pkg-2:nx-release-publish
+
+
+📦  @myorg/pkg-2@0.0.2
+=== Tarball Contents ===
+
+233B README.md
+277B package.json
+53B  src/index.ts
+61B  src/lib/pkg-2.ts
+=== Tarball Details ===
+name:          @myorg/pkg-2
+version:       0.0.2
+filename:      testorg-pkg-2-0.0.2.tgz
+package size:  531 B
+unpacked size: 624 B
+shasum:        {shasum}
+integrity:     {integrity}
+total files:   12
+
+Published to https://registry.npmjs.org with tag "latest"
+
+> nx run pkg-3:nx-release-publish
+
+
+📦  @myorg/pkg-3@0.0.2
+=== Tarball Contents ===
+
+233B README.md
+277B package.json
+53B  src/index.ts
+61B  src/lib/pkg-3.ts
+=== Tarball Details ===
+name:          @myorg/pkg-3
+version:       0.0.2
+filename:      testorg-pkg-3-0.0.2.tgz
+package size:  531 B
+unpacked size: 624 B
+shasum:        {shasum}
+integrity:     {integrity}
+total files:   12
+
+Published to https://registry.npmjs.org with tag "latest"
+
+ —————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
+
+ >  NX   Successfully ran target nx-release-publish for 3 projects
+```
+
+## Publish in Github Actions
+
+A common way to automate publishing packages is via Github Actions. An example of a publish workflow is as follows:
+
+```yaml
+# ./.github/workflows/publish.yml
+name: Publish
+
+on:
+  push:
+    tags:
+      - v*.*.*
+
+jobs:
+  test:
+    name: Publish
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write # needed for provenance data generation
```

Comment:
In trying this today I noticed you need to add back the `contents: read` permission as well. I think setting any granular permissions removes the defaults